src/tests/portpicker_test.py

```diff
@@ -92,6 +95,50 @@ def testPidDefaultsToOwnPid(self):
                 server.sendall.assert_called_once_with(b'9876\n')
         self.assertEqual(port, 52768)
 
+    @mock.patch.dict(os.environ,{'PORTSERVER_ADDRESS': 'portserver'})
+    def testReusesPortServerPorts(self):
+        server = mock.Mock()
+        server.recv.side_effect = [b'12345\n', b'23456\n', b'34567\n']
+        with mock.patch.object(socket, 'socket', return_value=server):
+            self.assertEqual(portpicker.pick_unused_port(), 12345)
+            self.assertEqual(portpicker.pick_unused_port(), 23456)
+            portpicker.return_port(12345)
+            self.assertEqual(portpicker.pick_unused_port(), 12345)
+
+    @mock.patch.dict(os.environ,{'PORTSERVER_ADDRESS': ''})
+    def testDoesntReuseRandomPorts(self):
+        ports = set()
+        for _ in range(10):
+            port = portpicker.pick_unused_port()
+            ports.add(port)
+            portpicker.return_port(port)
+        self.assertGreater(len(ports), 5)  # Allow some random reuse.
```

Comment:
This _might_ turn out to be flaky in the future... but the chances of the rng choosing the same port out of 10000 choices 5 times in a row seem extremely low so lets just keep this as is for now until it becomes an issue (if ever).